tlair/src/main/scala/at/forsyte/apalache/tla/lir/oper/ApalacheOper.scala

```diff
@@ -61,6 +61,22 @@ object ApalacheOper {
     override def precedence: (Int, Int) = (100, 100)
   }
 
+  /**
+   * A non-deterministic version of `CHOOSE x \in S: TRUE`. Since we cannot define a new syntactic form where `x` ranges
```

Comment:
I think direct definitions, not relative to another operator, are easier to understand.
```suggestion
   * Non-deterministically pick a value out of the set `S`, if `S` is non-empty. If `S` is empty, return some value of the proper type.
   * This can be understood as a non-deterministic version of `CHOOSE x \in S: TRUE`. Since we cannot define a new syntactic form where `x` ranges
```